test/node/parsing/HTMLFileContent-test.js

```diff
@@ -0,0 +1,84 @@
+// Copyright 2017 The Chromium Authors. ALl rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+describe('HTMLFileContent', function() {
+  var HTMLFileContent;
+  var Parser;
+
+  function cmpTest(testName, testDirectory, expectedPre, expectedIDL) {
+    var fs = require('fs');
+    var spec = fs.readFileSync(`${testDirectory}/spec.html`).toString();
+    var htmlSpec = HTMLFileContent.create({ file: spec });
+    var preBlocks = htmlSpec.pre;
+    var idlFragments = preBlocks.filter(function(block) {
+      return block.isIDL;
+    });
+
+    // Determine the number of fragments that were found
+    expect(preBlocks.length).toBe(expectedPre)
+    expect(idlFragments.length).toBe(expectedIDL);
+
+    fs.readdir(testDirectory, function(err, files) {
+      // Go through each of the expected results in the folder
+      files.forEach(function(filename) {
+        var testNum = Number(filename);
+
+        if (!isNaN(testNum) && testNum < preBlocks.length) {
+          var expectedContent = fs.readFileSync(`${testDirectory}/${filename}`).toString();
+          expect(preBlocks[testNum].content.trim()).toBe(expectedContent.trim());
+        } else if (filename !== 'spec.html') {
+          console.warn(`${filename} was not used in ${testName} spec test`);
+        }
+      });
+    });
+  }
+
+  beforeEach(function() {
+    HTMLFileContent = foam.lookup('org.chromium.webidl.HTMLFileContent');
+  });
+
+  it('should parse a pre tag with no content', function() {
+    var content = '<pre class="idl"></pre>';
```

Comment:
Whoops, this was actually one of the old test that I am no longer using. Will likely modify this test or remove in next commit